src_c/event.c

```diff
@@ -1901,6 +1919,29 @@ _pg_get_all_events(void)
         for (loop = 0; loop < len; loop++) {
             if (!_pg_event_append_to_list(list, &eventbuf[loop]))
                 goto error;
+
+            if (eventbuf[loop].type == SDL_KEYDOWN) {
+                int i;
+                for (i = 0; i < NUM_KEYS; i++) {
+                    if (i == (int)eventbuf[loop].key.keysym.scancode &&
+                        !keys_being_held_down[i]) {
+                        held_down_keys[i] = 1;
+                        keys_being_held_down[i] = 1;
+                    }
+                }
+            }
```

Comment:
If my understanding of how this works is correct, isn't this for loop completely unnecessary?

You can just index using the scancode of the event directly?

Reply:
Also isn't `keys_being_held_down` unnecessary information now? If you get a keydown, it means it was just pressed. If you get a keyup, it means it was just released.

Reply:
Thanks for this, yeah you are right. Didn't think of removing the for loop after I stopped using SDL_GetKeyboardState